cfe_internal/enterprise/federation/federation.cf

```diff
@@ -530,18 +530,40 @@ bundle agent federation_manage_files
 
 bundle agent postgres_config
 # @brief Customize postgres config for superhub
+# @variable cfengine_enterprise_federation.postgres_config.shared_buffers - Sets the maximum number of locks per transaction.
+# @variable cfengine_enterprise_federation.postgres_config.max_locks_per_transaction - Sets the maximum number of locks per transaction.
+# @variable cfengine_enterprise_federation.postgres_config.log_lock_waits - Logs long lock waits.
+# @variable cfengine_enterprise_federation.postgres_config.max_wal_size - Sets the WAL size that triggers a checkpoint.
+# @variable cfengine_enterprise_federation.postgres_config.checkpoint_timeout- Sets the maximum time between automatic WAL checkpoints.
 {
   vars:
     am_superhub::
-      "c[shared_buffers]"
-        string => "1GB",
+      "c[shared_buffers]" -> { "ENT-8617" }
+        string => ifelse( isvariable( "cfengine_enterprise_federation.postgres_config.shared_buffers"),
+                          $(cfengine_enterprise_federation.postgres_config.shared_buffers),
+                          "1GB"),
         comment => "Changing this setting requires restarting the database.";
-      "c[max_locks_per_transaction]"
-        string => "4000",
+      "c[max_locks_per_transaction]" -> { "ENT-8617" }
+        string => ifelse( isvariable( "cfengine_enterprise_federation.postgres_config.max_locks_per_transaction"),
+                          $(cfengine_enterprise_federation.postgres_config.max_locks_per_transaction),
+                          "4000"),
         comment => "Changing this setting requires restarting the database.";
-      "c[log_lock_waits]"
-        string => "on",
+      "c[log_lock_waits]" -> { "ENT-8617" }
+        string => ifelse( isvariable( "cfengine_enterprise_federation.postgres_config.log_lock_waits"),
+                          $(cfengine_enterprise_federation.postgres_config.log_lock_waits),
+                          "on"),
         comment => "Changing this setting requires restarting the database.";
+
+      "c[max_wal_size]" -> { "ENT-8617" }
+        string => ifelse( isvariable( "cfengine_enterprise_federation.postgres_config.max_wal_size"),
+                          $(cfengine_enterprise_federation.postgres_config.max_wal_size),
+                          "1GB");
+
+      "c[checkpoint_timeout]" -> { "ENT-8617" }
+        string => ifelse( isvariable( "cfengine_enterprise_federation.postgres_config.checkpoint_timeout"),
+                          $(cfengine_enterprise_federation.postgres_config.checkpoint_timeout),
+                          "5min");
```

Comment:
I really don't know what a good value for this is, given the un-predictable nature of when feeders dump and what cycle the checkpoint timeout might be in and whether wal gets exhausted :( Maybe something higher like 30min or 40min with the idea that we want to avoid automatic checkpoints during EVERY import cycle? I'm not sure. I suppose ideally a checkpoint might occur immediately AFTER an import actually, but postgres docs, https://www.postgresql.org/docs/current/sql-checkpoint.html, mention not to use that in "normal operations".

Reply:
Since these tunables are only relevant in the context of federated reporting superhub I figured there was no need to mention that. None of these tunables will be applied if the hub is not enabled for federated reporting.

I figure if we want to suggest better defaults, we should instead change our defaults in policy. The hint that I left here is that the augments example values are using the values that we have recently suggested setting.